### ISSUE TITLE
Implement Simple-HHEA as an EA BaseLinker (prerequisite for #22's llama2-13b comparison)

### ISSUE THREAD

**Issue by jmccrae** [jmccrae/linkingtk#62]
While planning #22 (ChatEA), found the paper's llama2-13b numbers (Hits@1=0.455/MRR=0.553 on ICEWS-WIKI, from Table 4 of https://aclanthology.org/2024.acl-long.408.pdf) are ChatEA layered on the paper's own base embedding method, Simple-HHEA -- not our existing `KGELinker` (plain TransE-family). Using `KGELinker` as ChatEA's base would confound any comparison to the published number with candidate-quality differences unrelated to the ChatEA-reasoning port itself.

Implement `SimpleHHEALinker` as its own EA `BaseLinker`, porting the reference implementation at https://github.com/DataArcTech/Simple-HHEA (paper: "Toward Practical Entity Alignment Method Design: Insights from New Highly Heterogeneous Knowledge Graph Datasets", Jiang et al. 2024). Three fused feature groups: ALBERT name embeddings + whitening, node2vec structural embeddings, Time2Vec temporal embeddings (ICEWS only) -- trained end-to-end with margin ranking loss, evaluated via CSLS (already implemented in `linkingtk.eval.ranking`).

Target dataset: ICEWS-WIKI only (`linkingtk.datasets.icews.IcewsWikiDataset` already exists and fetches this). ICEWS-YAGO stays out of scope for now (split .z01/.z02/.zip archive, unsupported by the current single-zip fetch path).

Once closed, #22 (ChatEA) will use `SimpleHHEALinker` as its base candidate-generation method for a genuine llama2-13b comparison.

**Comment by jmccrae** [comment]
Closed with a real ICEWS-WIKI run (commit 8c38ce1, CI green: https://github.com/jmccrae/linkingtk/actions/runs/32962082985):

```
1518 train / 3540 test pairs
Metrics: {'Hits@1': 0.262, 'Hits@10': 0.477, 'MRR': 0.337}
```

vs. the paper's own published Simple-HHEA number for ICEWS-WIKI (Hits@1=0.720, Hits@10=0.872, MRR=0.754, Table 2). Full writeup: docs/examples/simple_hhea_ea.md.

Two real bugs found and fixed along the way (both covered by regression tests):
- **node2vec never finished** on ICEWS-WIKI's real graph -- measured directly: median node degree 18, but one node has 273,317 edges. node2vec's alias-table precompute is O(degree) per edge, O(degree^2) for a hub node -- confirmed stuck for many CPU-bound minutes with zero progress before being killed. Fixed by adding `cap_node_degree`, which randomly subsamples any node's excess edges down to a `max_degree` cap (default 1000) before node2vec ever sees the graph.
- **A `PYTHONHASHSEED`-driven non-determinism bug**: deduplicating edges via a plain Python `set` before capping meant the seeded `rng.permutation` in `cap_node_degree` was shuffling indices into a hash-order-dependent list, so the same `random_state` picked different actual edges across separate process runs. Caught because it made a fully-seeded unit test flaky across repeated runs. Fixed by sorting the deduplicated edge list before capping.

Also corrected the benchmark's evaluation metric (`inner` -> `cosine`) to match the reference's own `CSLS_.py`, which L2-normalizes both embedding matrices before its raw `matmul` -- that alone moved Hits@1 from 0.237 to 0.262.

**The remaining gap is attributed to `max_degree` itself and reported honestly, not chased further**: capping ICEWS-WIKI's highest-degree entities from up to 273,317 neighbors down to 1000 throws away real structural signal for exactly the entities a power-law graph concentrates the most information in. The reference paper never discusses this scaling problem at all, plausibly because a research run had a much larger compute/time budget than a demo script here does for a single node2vec precompute pass. Raising `max_degree` was checked as directly costly (precompute cost grows roughly with `max_degree^2` for capped hub nodes), not assumed cheap, and left as a documented, tunable tradeoff.

`SimpleHHEALinker` is now available as a real embedding base for #22 (ChatEA)'s `base_linker`, for a genuine comparison against the ChatEA paper's own published llama2-13b number on ICEWS-WIKI.

**Comment by jmccrae** [comment]
**Update: the closing numbers above were superseded by a real bug fix.**

The original close used the third-party `node2vec` PyPI package for structural embeddings, reusing the reference's literal `p=1e-100` hyperparameter. That package implements the *standard* Grover-Leskovec divide-by-`p` alias-edge formula, but the reference's own `longterm/node2vec.py` uses a **non-standard** multiply-by-`p` formula instead (the standard version is literally commented out directly above it in their source — a deliberate inversion). Reusing the reference's `p` value with the standard-convention package therefore did the *opposite* of the intended "never immediately backtrack," producing degenerate two-node-oscillation walks.

Hand-ported the reference's exact alias-sampling formula instead of continuing to approximate it (`_alias_setup`/`_alias_draw`/`_get_alias_edge`/`_node2vec_walk`/`_simulate_walks` in `_simple_hhea_structure.py`); the third-party `node2vec` dependency is removed in favor of `networkx`+`gensim` directly.

Surprising result: even the faithfully-ported reference formula scores *worse* on ICEWS-WIKI (Hits@1=0.015) than the original buggy version (0.262). Diagnosed with a direct ablation — disabling the structural branch entirely (`use_structure=False`, name+time embeddings only) beats every structure-enabled variant tried, including the paper's own published number:

```
1518 train / 3540 test pairs
Hits@1=0.8147, Hits@10=0.8822, MRR=0.8384
```

vs. paper's published Simple-HHEA on ICEWS-WIKI: Hits@1=0.720, Hits@10=0.872, MRR=0.754.

Likely cause: ICEWS-WIKI is deliberately the paper's "highly heterogeneous" stress case — the ICEWS side is a dense political-event network, the Wikipedia side is a hyperlink graph, and only the 1,518 train-pair-merged nodes tie them together at all, so a node2vec embedding trained mostly within one side has little reason to be cross-domain comparable by cosine similarity. The name branch doesn't have this problem since it comes from one shared pretrained language model.

`examples/simple_hhea_ea.py` now uses `use_structure=False` for this dataset specifically (with the full evidence trail in `docs/examples/simple_hhea_ea.md`); `SimpleHHEALinker`'s own constructor default stays `use_structure=True`, since it's faithful to the reference and may still help on more structurally-homogeneous datasets.

All verification re-run clean (mypy --strict, ruff, full pytest suite — 787 passed, mkdocs build --strict). Pushed in 68a5161.
